environment.yml

```diff
@@ -20,7 +20,7 @@ dependencies:
   # code checks
   - black=20.8b1
   - cpplint
-  - flake8
+  - flake8=3.9.0
```

Comment:
can we have 1 build where this is unpinned?

Reply:
Sure, but there's the risk that there's be another annoying CI failure like #40457 

If there's a new flake8 release, this'll get bumped up anyway by the weekly autoupdate job

Reply:
yeah this is ok then.